lib/share_progress/variation.rb

```diff
@@ -0,0 +1,92 @@
+require 'share_progress'
+require 'share_progress/client'
+
+module ShareProgress
+  class Variation
+    class << self
+      def facebook_type_name
+        'facebook'
+      end
+
+      def twitter_type_name
+        'twitter'
+      end
+
+      def email_type_name
```

Comment:
what is this for? should these each be the `type_name` method of the respective classes?